kernel/constant_typing.ml

```diff
@@ -28,7 +28,9 @@ module NamedDecl = Context.Named.Declaration
 (* Checks the section variables for the body.
    Returns the closure of the union with the variables in the type.
 *)
-let check_section_variables env declared_set typ body =
+let check_section_variables env declared_set body typ =
+  let env_ids = ids_of_named_context_val (named_context_val env) in
+  Id.Set.iter (fun id -> if not (Id.Set.mem id env_ids) then user_err Pp.(str "Unknown variable " ++ Id.print id)) declared_set;
```

Comment:
You should add an entry to type_errors rather than printing it directly here.

Reply:
OK, will do.